ivy/functional/frontends/tensorflow/general_functions.py

```diff
@@ -115,6 +115,41 @@ def foldl(
     return result
 
 
+@to_ivy_arrays_and_back
+def foldr(
+    fn,
+    elems,
+    initializer=None,
+    parallel_iterations=10,
+    back_prop=True,
```

Comment:
why can't we use `back_prop`

Reply:
Not sure how to implement it. Tried using `ivy.execute_with_gradients()` with `retain_grads=True`. However not sure if it serves the same functionality. Let me know what I can do about that!